src/Frame/Frame.cc

```diff
@@ -2181,4 +2189,292 @@ void Frame::CloseCachedImage(const std::string& file) {
     }
 }
 
+bool Frame::SetVectorOverlayParameters(const CARTA::SetVectorOverlayParameters& message) {
+    VectorFieldSettings new_settings = {(int)message.smoothing_factor(), message.fractional(), message.threshold(), message.debiasing(),
+        message.q_error(), message.u_error(), message.stokes_intensity(), message.stokes_angle(), message.compression_type(),
+        message.compression_quality()};
+
+    if (_vector_field_settings != new_settings) {
+        _vector_field_settings = new_settings;
+        return true;
+    }
+    return false;
+}
+
+bool Frame::VectorFieldImage(VectorFieldCallback& partial_vector_field_callback) {
```

Comment:
Not compulsory at all, but It almost looks like this could be moved to an isolated struct/class (VectorField?), is that possible? I think that would make its purpose more readable than inside Frame.